### PR TITLE
feat(protocol-designer): deck setup skeleton for redesign

### PR DESCRIPTION
**Pull request by jerader**
closes AUTH-635

# Overview

This PR creates a skeleton for the deck setup in the redesign. It is not entirely wired up yet but you should be able to see the deck map and when you hover on the slots, it will show a "slot description" on the left. 

https://github.com/user-attachments/assets/cc6098b8-da57-484c-9147-a6ae693b6ba2

## Test Plan and Hands on Testing

First, with the redesign ff turned off, create a flex protocol and add some labware, modules and liquids to the deck. 
Then, turn on the redesign ff and click on "create new protocol". Then click on the "protocol overview" text. You should see a "go to deck setup" button. Click on that and see the deck map view. When you hover over each slot, it should give you the select overlay and a "slot information" text to the left. (see the attached video for an example)

Repeat the steps for an ot-2 protocol. should work the same.

Note: i haven't wired up select yet!! 

## Changelog

- create new `DeckSetup` components under the `ProtocolOverview` pages
- add a test case to `ProtocolOverview`

## Review requests

see test plans

## Risk assessment

low, behind ff

### CODE REVIEW

File: protocol-designer/src/localization/en/protocol_overview.json [hunk 1]
```diff
@@ -1,3 +1,5 @@
 {
-  "protocol_overview": "Protocol overview"
+  "protocol_overview": "Protocol overview",
+  "edit": "Edit",
+  "add_labware": "Add labware"
```
Comment (koji) [inline protocol-designer/src/localization/en/protocol_overview.json]:
```suggestion
  "add_labware": "Add labware",
  "edit": "Edit",
  "protocol_overview": "Protocol overview"
```

File: protocol-designer/src/pages/ProtocolOverview/DeckSetup/ControlSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        css={css`
+          opacity: ${hover != null && hover === slotId ? `1` : `0`};
+        `}
+      >
+        <a
+          onClick={() => {
+            addEquipment(slotId)
+          }}
+        >
+          <LegacyStyledText>
```
Comment (koji) [inline protocol-designer/src/pages/ProtocolOverview/DeckSetup/ControlSelect.tsx:82]: Don't need to use `StyledText`?

Comment (jerader) [reply protocol-designer/src/pages/ProtocolOverview/DeckSetup/ControlSelect.tsx:82]: this PR is just the skeleton of the deck setup experience. All these components will change still!

File: protocol-designer/src/pages/ProtocolOverview/DeckSetup/index.tsx [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+  const selectedTerminalItemId = useSelector(getSelectedTerminalItemId)
+  const activeDeckSetup = useSelector(getDeckSetupForActiveItem)
+  const _disableCollisionWarnings = useSelector(getDisableModuleRestrictions)
+  const trash = Object.values(activeDeckSetup.additionalEquipmentOnDeck).find(
+    ae => ae.name === 'trashBin'
+  )
+  const robotType = useSelector(getRobotType)
+  const deckDef = React.useMemo(() => getDeckDefFromRobotType(robotType), [])
+  const [hover, setHover] = React.useState<DeckSlot | null>(null)
+  const [zoomIn, setZoomInOnSlot] = React.useState<OpenSlot | null>(null)
```
Comment (koji) [inline protocol-designer/src/pages/ProtocolOverview/DeckSetup/index.tsx]:
```suggestion
  const robotType = useSelector(getRobotType)
  const deckDef = React.useMemo(() => getDeckDefFromRobotType(robotType), [])
  const [hover, setHover] = React.useState<DeckSlot | null>(null)
  const [zoomIn, setZoomInOnSlot] = React.useState<OpenSlot | null>(null)
  const trash = Object.values(activeDeckSetup.additionalEquipmentOnDeck).find(
    ae => ae.name === 'trashBin'
  )
```

File: protocol-designer/src/pages/ProtocolOverview/DeckSetup/index.tsx [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+          maxWidth={robotType === FLEX_ROBOT_TYPE ? '130vh' : '100vh'}
+          width="100%"
+          maxHeight="120vh"
+        >
+          <RobotCoordinateSpaceWithRef
+            height="90%"
+            deckDef={deckDef}
+            viewBox={`${deckDef.cornerOffsetFromOrigin[0]} ${
+              hasWasteChute
+                ? deckDef.cornerOffsetFromOrigin[1] - 30
```
Comment (koji) [inline protocol-designer/src/pages/ProtocolOverview/DeckSetup/index.tsx]: add a new const for `30`?

**Review by koji — APPROVED** [review]
lgtm

**Pull request merged**